x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/task/rule_migrations_task_client.ts

```diff
@@ -109,19 +114,30 @@ export class RuleMigrationsTaskClient {
   /** Returns the stats of a migration */
   public async getStats(migrationId: string): Promise<RuleMigrationTaskStats> {
     const dataStats = await this.data.rules.getStats(migrationId);
-    const status = this.getTaskStatus(migrationId, dataStats.rules);
-    return { status, ...dataStats };
+    const taskStats = this.getTaskStats(migrationId, dataStats.rules);
+    return { ...taskStats, ...dataStats };
   }
 
   /** Returns the stats of all migrations */
   async getAllStats(): Promise<RuleMigrationTaskStats[]> {
     const allDataStats = await this.data.rules.getAllStats();
     return allDataStats.map((dataStats) => {
-      const status = this.getTaskStatus(dataStats.id, dataStats.rules);
-      return { status, ...dataStats };
+      const taskStats = this.getTaskStats(dataStats.id, dataStats.rules);
+      return { ...taskStats, ...dataStats };
     });
   }
 
+  private getTaskStats(
+    migrationId: string,
+    dataStats: RuleMigrationDataStats['rules']
+  ): Pick<RuleMigrationTaskStats, 'status' | 'last_error'> {
+    const lastError = RuleMigrationsTaskClient.migrationsLastError.get(migrationId);
+    return {
+      status: this.getTaskStatus(migrationId, dataStats),
+      ...(lastError && { last_error: lastError.message }),
```

Comment:
I think this is a risky syntax since `lastError` can be `undefined` and spreading `undefined` can result in error in most cases. I am not sure why it does not fail in object 🤯 .

```js
console.log(...undefined) // Error: can't access property Symbol.iterator of undefined
console.log([...undefined]) // Error: can't access property Symbol.iterator of undefined
console.log({...undefined}) // works somehow 🤷 

```

Do you think below works better?
```suggestion
      lastError : lastError?.message,
```

Reply:
Well, yes this is a short-circuit evaluation, it's the same thing as doing 
```ts
...(lastError ? { last_error: lastError.message } : {}),
```
but more compact. 
Your suggestion would work, but it's not the same thing, it always sets the `lastError` key. I wanted to not even set the key if there's no last error.

And yes, spreading `undefined` or `null` in an object is a valid syntax in ES6, they are just ignored.